proofread deployment docs (#1308)
emqtt/emq-docs-cn: en_US/deploy/cluster/db.md

- etc.

Mria tables are replicated across all EMQX nodes.
It also helps with the fault-tolerance: the data is safe as long as at least one node in the cluster is alive.

> If the size of the EMQX cluster is below 5 nodes, typically, no database scalability tuning is required.

Reviewer: I think these commas are not necessary, English is very lax about commas

Author: AI checker always wants commas around introductory phrases 😅

This approach solves two problems:

- Horizontal scalability (we've tested EMQX cluster with 23 nodes)

- It enables autoscaling of the replicant cluster

Since replicant nodes don't participate in writes, the efficiency of table updates doesn't suffer when more replicants are added to the cluster.
This allows the creation of larger EMQX clusters.

Also, replicant nodes are designed to be ephemeral.
Adding or removing them won't change the data redundancy, so they can be placed in an autoscaling group, thus enabling better DevOps practices.
Note that initial replication of the data from the core nodes is a relatively heavy operation, depending on the data size, so the autoscaling policy must not be very aggressive.

## Configuration

> In EMQX 5.0 all nodes assume a core role by default, so without tweaks, the cluster behaves like 4.*.

Reviewer: It feels like article is not necessary here

Author: ✅

> In EMQX 5.0 all nodes assume the core role by default, so without tweaks, the cluster behaves like 4.*.

Reviewer 2: ```suggestion
In EMQX 5.0 all nodes assume a core role by default, so without tweaks, the cluster behaves like 4.*.
```
?

> To use the new replication protocol, set `EMQX_NODE__DB_ROLE` environment variable or `node.db_role` setting in `emqx.conf` to `replicant` on some of the nodes in the cluster. This way, they will assume the replicant role. Note that there must be at least one core node in the cluster. We recommend 3 cores + N replicants setup as the starting point.

Reviewer 2: ```suggestion
To use the new replication protocol, set `EMQX_NODE__DB_ROLE` environment variable or `node.db_role` setting in `emqx.conf` to `replicant` on some of the nodes in the cluster. This way, they will assume a replicant role. Note that there must be at least one core node in the cluster. We recommend 3 cores + N replicants setup as the starting point.
```
?

Core nodes may accept MQTT traffic, or they can disable all MQTT listeners to serve purely as the database server for the replicants, depending on the use case.

> - In a small cluster (3 nodes or less in total) it doesn't make economic sense to use replicants, so core nodes take all the traffic.

Reviewer 2: Maybe:

```suggestion
- In a small cluster (3 nodes or less in total) it doesn't make sense perfomance-wise to use replicants, so core nodes take all the traffic.
```
?

- In a very large cluster (10 nodes or more) it makes sense to move away traffic from the core nodes.
- In a medium cluster it really depends on many factors, so experimentation is needed.

## Monitoring and troubleshooting

The Mria performance can be monitored using Prometheus metrics or Erlang console.

### Prometheus metrics

#### Core
- `emqx_mria_last_intercepted_trans`: Number of transactions received by the shard since the node started. Note that this value can be different on different core nodes.
- `emqx_mria_weight`: A value used for load balancing. It changes depending on the momentary load of the core node.
- `emqx_mria_replicants`: Number of replicants connected to the core node. Numbers are grouped per shard.

> - `emqx_mria_server_mql`: Number of unprocessed transactions waiting to be sent to the replicants. Less is better. If this metric grows, then it's probably time to add more computing resources for the exiting core nodes or even add more core nodes.

Reviewer: grows

Author: 🤔
